common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -683,50 +683,64 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
-
-            for answer_id, orig_answers in lcp.student_answers.items():
-                # Some types of problems have data in lcp.student_answers that isn't in lcp.problem_data.
-                # E.g. formulae do this to store the MathML version of the answer.
-                # We exclude these rows from the report because we only need the text-only answer.
-                if answer_id.endswith('_dynamath'):
-                    continue
-
-                if limit_responses and count >= limit_responses:
-                    # End the iterator here
-                    return
-
-                question_text = lcp.find_question_label(answer_id)
-                answer_text = lcp.find_answer_text(answer_id, current_answer=orig_answers)
-                correct_answer_text = lcp.find_correct_answer_text(answer_id)
-
-                count += 1
+                for answer_id, orig_answers in lcp.student_answers.items():
+                    # Some types of problems have data in lcp.student_answers that isn't in lcp.problem_data.
+                    # E.g. formulae do this to store the MathML version of the answer.
+                    # We exclude these rows from the report because we only need the text-only answer.
+                    if answer_id.endswith('_dynamath'):
+                        continue
+
+                    if limit_responses and count >= limit_responses:
+                        # End the iterator here
+                        return
+
+                    question_text = lcp.find_question_label(answer_id)
+                    answer_text = lcp.find_answer_text(answer_id, current_answer=orig_answers)
+                    correct_answer_text = lcp.find_correct_answer_text(answer_id)
+
+                    count += 1
+                    report = {
+                        _("Answer ID"): answer_id,
+                        _("Question"): question_text,
+                        _("Answer"): answer_text,
+                    }
+                    if correct_answer_text is not None:
+                        report[_("Correct Answer")] = correct_answer_text
+                    yield (user_state.username, report)
+            except LoncapaProblemError:
+                # Capture a backtrace for errors from failed loncapa problems
+                log.exception(
+                    "An error occured generating a problem report on course %s, problem %s, and student %s",
+                    self.course_id, self.scope_ids.usage_id,
+                    self.scope_ids.user_id
+                )
+                # Also input error in report
                 report = {
-                    _("Answer ID"): answer_id,
-                    _("Question"): question_text,
-                    _("Answer"): answer_text,
+                    _("Answer ID"): "Python Error",
+                    _("Question"): "Attempting to generate a report on the problem failed.",
+                    _("Answer"): "Python Error: No Answer Retrieved",
```

Comment:
I don't remember the format of this report. Is it possible to trace back from this to what user and problem this failed at, so we can investigate further when something goes wrong?

Reply:
This "report" object is the "state" section in the report. I have included a screencap of the start of a report below. It includes the "location," "username", "and block key" so instructors will be able to locate particular problems with indovidual student answers.

<img width="1138" alt="Screen Shot 2021-07-16 at 11 36 21 AM" src="https://user-images.githubusercontent.com/49422820/125972943-60bbc406-3131-4f4a-92c8-396bce8d3f5e.png">

Reply:
These other fields are separate from the "report" dict created